session.go

```diff
@@ -366,15 +383,20 @@ func (s *session) EnqueueBytesAndSend(msg []byte) {
 	s.sendQueued()
 }
 
-func (s *session) sendBytes(msg []byte) {
+func (s *session) sendBytes(msg []byte) bool {
 	if s.messageOut == nil {
 		s.log.OnEventf("Failed to send: disconnected")
-		return
+		return false
 	}
 
-	s.log.OnOutgoing(msg)
-	s.messageOut <- msg
-	s.stateTimer.Reset(s.HeartBtInt)
+	select {
+	case <-time.After(5 * time.Millisecond):
```

Comment:
while `time.After` is a nice convenience method, it shouldn't be used in hotpaths where you call it a many times as internally it creates a timer but never stops it, so you should rather just create a timer yourself so you can stop it if you end up returning before it fires.

it is also documented as such: https://pkg.go.dev/time#After

(although I also commented above that I think maybe there shouldn't be a timer at all and just use a default?)

Reply:
Addressed: https://github.com/alpacahq/quickfix/pull/18/commits/21384e6dc24b832a5b3613f534b1256d6d4ebf85